akka-remote/src/main/scala/akka/remote/artery/TestStage.scala

```diff
@@ -173,11 +172,24 @@ private[remote] class InboundTestStage(inboundContext: InboundContext, state: Sh
                   push(out, env)
               case _ =>
                 // unknown, handshake not completed
-                if (state.anyBlackholePresent())
-                  log.debug(
-                    "inbound message [{}] before handshake completed, cannot check if remote is blackholed, letting through",
-                    Logging.messageClassName(env.message))
-                push(out, env)
+                if (state.anyBlackholePresent()) {
+                  env.message match {
+                    case _: HandshakeReq =>
+                      log.debug(
+                        "inbound message [{}] before handshake completed, cannot check if remote is blackholed, letting through",
+                        Logging.messageClassName(env.message))
+                      push(out, env) // let it through
+
+                    case anyOther =>
+                      log.debug(
+                        "dropping inbound message [{}] with UID [{}] because of blackhole",
+                        Logging.messageClassName(anyOther),
+                        env.originUid)
+                      pull(in) // drop message
+                  }
+                } else {
+                  push(out, env)
+                }
```

Comment:
At least for the scenario where we had problems (gossip coming through) I think this should fix it. 👍